grid2op/tests/test_timeOutEnvironment.py

```diff
@@ -0,0 +1,149 @@
+    def get_timeout_ms(self):
+        return 100
+        
+    def setUp(self) -> None:
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore")
+            # TODO : Comment on fait avec un time out ?
+            self.env1 = TimedOutEnvironment(grid2op.make("l2rpn_case14_sandbox", test=True),
+                                            time_out_ms=self.get_timeout_ms())
+        params = self.env1.parameters
+        params.NO_OVERFLOW_DISCONNECTION = True
+        self.env1.change_parameters(params)
+
+    def test_no_dn(self):
+        agentok = AgentOK(self.env1)
+        obs = self.env1.reset()
+        assert self.env1._nb_dn_last == 0
+        for i in range(10):
+            act = agentok.act(None, None, None)
+            obs, reward, done, info = self.env1.step(act)
+            assert info["nb_do_nothing"] == 0
+            assert info["nb_do_nothing_made"] == 0
+            assert self.env1._nb_dn_last == 0
+            
+    def test_one_dn(self):
+        agentko = AgentKO(self.env1)
+        obs = self.env1.reset()
+        assert self.env1._nb_dn_last == 0
+        for i in range(10):
+            act = agentko.act(None, None, None)
+            obs, reward, done, info = self.env1.step(act)
+            assert info["nb_do_nothing"] == 1
+            assert info["nb_do_nothing_made"] == 1
+            assert self.env1._nb_dn_last == 1
+            
+    def test_one_dn2(self):
+        agentko = AgentKO1(self.env1)
+        obs = self.env1.reset()
+        assert self.env1._nb_dn_last == 0
+        for i in range(10):
+            act = agentko.act(None, None, None)
+            obs, reward, done, info = self.env1.step(act)
+            assert info["nb_do_nothing"] == 1
+            assert info["nb_do_nothing_made"] == 1
+            assert self.env1._nb_dn_last == 1
+
+    def test_two_dn(self):
+        agentko2 = AgentKO2(self.env1)
+        obs = self.env1.reset()
+        assert self.env1._nb_dn_last == 0
+        for i in range(10):
+            act = agentko2.act(None, None, None)
+            obs, reward, done, info = self.env1.step(act)
+            assert info["nb_do_nothing"] == 2
+            assert info["nb_do_nothing_made"] == 2
+            assert self.env1._nb_dn_last == 2
+    
+    def test_diff_dn(self):
+        agentok = AgentOK(self.env1)
+        agentko = AgentKO(self.env1)
+        agentko2 = AgentKO2(self.env1)
+        obs = self.env1.reset()
+        assert self.env1._nb_dn_last == 0
+        for i, agent in enumerate([agentok, agentko, agentko2] * 2):
+            act = agent.act(None, None, None)
+            obs, reward, done, info = self.env1.step(act)
+            assert info["nb_do_nothing"] == i % 3
+            assert info["nb_do_nothing_made"] == i % 3
+            assert self.env1._nb_dn_last == i % 3
+        
+
+class TestTimedOutEnvironment50(TestTimedOutEnvironment100):
+    def get_timeout_ms(self):
+        return 50
```

Comment:
make that 150 to make the test pass due to the limitation of the "sleep" function